server/src/main/java/org/opensearch/index/translog/InternalTranslogManager.java

```diff
@@ -279,6 +283,26 @@ public void ensureCanFlush() {
         }
     }
 
+    /**
+     * Reads operations from the translog
+     * @param location
+     * @return
+     * @throws IOException
+     */
+    public Translog.Operation readOperation(Translog.Location location) throws IOException {
+        return translog.readOperation(location);
+    }
+
+    /**
+     * Adds operations to the translog
+     * @param operation
+     * @return
+     * @throws IOException
+     */
+    public Translog.Location add(Translog.Operation operation) throws IOException {
```

Comment:
Shouldn't these methods part of `TranslogManager` interface? These seem very basic translog operations.